internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -1115,6 +1307,64 @@ func NewGrafanaContainer(cr *model.CryostatInstance, imageTag string, tls *TLSCo
 	}
 }
 
+/*
+ * storage.image.repository - Repository for the database container
+ * storage.image.pullPolicy - Pull Policy for the database container
+ * storage.image.tag - Image tag for the database container
+ * storage.resources - Resource requests/limits for the database container
+ * storage.securityContext - Security Context for the database container
+ */
+func NewStorageContainer(cr *model.CryostatInstance, imageTag string, tls *TLSConfig) corev1.Container {
+	var containerSc *corev1.SecurityContext
+	if cr.Spec.SecurityOptions != nil && cr.Spec.SecurityOptions.StorageSecurityContext != nil {
+		containerSc = cr.Spec.SecurityOptions.StorageSecurityContext
+	} else {
+		privEscalation := false
+		containerSc = &corev1.SecurityContext{
+			AllowPrivilegeEscalation: &privEscalation,
+			Capabilities: &corev1.Capabilities{
+				Drop: []corev1.Capability{constants.CapabilityAll},
+			},
+		}
+	}
+
+	return corev1.Container{
+		Name:            cr.Name + "-storage",
+		Image:           imageTag,
+		ImagePullPolicy: getPullPolicy(imageTag),
+		SecurityContext: containerSc,
+	}
+}
+
+/*
+ * db.image.repository - Repository for the database container
+ * db.image.pullPolicy - Pull Policy for the database container
+ * db.image.tag - Image tag for the database container
+ * db.resources - Resource requests/limits for the database container
+ * db.securityContext - Security Context for the database container
+ */
+func newDatabaseContainer(cr *model.CryostatInstance, imageTag string, tls *TLSConfig) corev1.Container {
+	var containerSc *corev1.SecurityContext
+	if cr.Spec.SecurityOptions != nil && cr.Spec.SecurityOptions.DatabaseSecurityContext != nil {
+		containerSc = cr.Spec.SecurityOptions.DatabaseSecurityContext
+	} else {
+		privEscalation := false
+		containerSc = &corev1.SecurityContext{
+			AllowPrivilegeEscalation: &privEscalation,
+			Capabilities: &corev1.Capabilities{
+				Drop: []corev1.Capability{constants.CapabilityAll},
+			},
+		}
+	}
+
+	return corev1.Container{
```

Comment:
I think this is missing some environment variables still: `POSTGRESQL_USER, POSTGRESQL_PASSWORD, POSTGRESQL_DATABASE, PG_ENCRYPT_KEY`. See https://github.com/cryostatio/cryostat3/blob/ef9d56b26f2df0da6ac75d82e048a765c732faa1/compose/db.yml#L14 for example. The user, pass, and database names all need to match with the values supplied to the core container. The encryption key should probably just be a randomly generated key, but it might make sense to have this configurable by the user as well. @ebaron WDYT?

Reply:
Having it be configurable would be nice, but I don't think strictly necessary for 3.0. Long term, I think we should handle it in the same manner as the password, where it's configurable. I think there's still a bug with the database password to be resolved though: https://github.com/cryostatio/cryostat-operator/issues/475.

Reply:
Speaking of which, is the [database password](https://github.com/cryostatio/cryostat-operator/blob/0b380d227f7b047571f8adf0a2467281ac3a5758/api/v1beta2/cryostat_types.go#L554-L560) still used?